pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -180,6 +183,59 @@ func (dst *EnvironmentResource) ConvertFrom(src v1.DataModelInterface) error {
 	return nil
 }
 
+func toRecipeConfigDatamodel(config *RecipeConfigProperties) datamodel.RecipeConfigProperties {
+	if config != nil {
+		recipeConfig := datamodel.RecipeConfigProperties{}
+		if config.Terraform != nil {
+			recipeConfig.Terraform = datamodel.TerraformConfigProperties{}
+			if config.Terraform.Authentication != nil {
+				recipeConfig.Terraform.Authentication = datamodel.AuthConfig{}
+				gitConfig := config.Terraform.Authentication.Git
+				if gitConfig != nil {
+					recipeConfig.Terraform.Authentication.Git = datamodel.GitAuthConfig{}
+					if gitConfig.Pat != nil {
+						p := map[string]datamodel.SecretConfig{}
+						for k, v := range gitConfig.Pat {
+							p[k] = datamodel.SecretConfig{
+								Secret: to.String(v.Secret),
+							}
+						}
+						recipeConfig.Terraform.Authentication.Git.PAT = p
+					}
+
+				}
+			}
+		}
+		return recipeConfig
+	}
+	return datamodel.RecipeConfigProperties{}
```

Comment:
```
	if config == nil || config.Terraform == nil || config.Terraform.Authentication == nil || config.Terraform.Authentication.Git == nil {
		return datamodel.RecipeConfigProperties{}
	}
```
I think it will be an issue when we have more properties in config e.g bicep config or when we add different modes of Auth ?

Reply:
I understand your point, but how do we solve that with the currently implemented reverse checks?

Reply:
Discussed offline, we are going to keep it as is to be able to support more properties in recipe config in the future.